official/vision/beta/modeling/backbones/mobilenet.py

```diff
@@ -0,0 +1,617 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Contains definitions of Mobilenet Networks."""
+
+from typing import Text, Optional, Dict
+
+# Import libraries
+import tensorflow as tf
+from official.vision.beta.modeling.backbones import factory
+from official.vision.beta.modeling.layers import nn_blocks
+from official.vision.beta.modeling.layers import nn_layers
+
+layers = tf.keras.layers
+regularizers = tf.keras.regularizers
+
+
+class GlobalPoolingBlock(tf.keras.layers.Layer):
+  def __init__(self, **kwargs):
+    super(GlobalPoolingBlock, self).__init__(**kwargs)
+
+  def call(self, inputs, training=None):
+    x = layers.GlobalAveragePooling2D()(inputs)
+    outputs = layers.Reshape((1, 1, x.shape[1]))(x)
+    return outputs
+
+
+"""
+Architecture: https://arxiv.org/abs/1704.04861.
+
+"MobileNets: Efficient Convolutional Neural Networks for
+  Mobile Vision Applications"
+Andrew G. Howard, Menglong Zhu, Bo Chen, Dmitry Kalenichenko, Weijun Wang,
+  Tobias Weyand, Marco Andreetto, Hartwig Adam
+"""
+MNV1_BLOCK_SPECS = {
+    'spec_name': 'MobileNetV1',
+    'block_spec_schema': ['block_fn', 'kernel_size', 'strides', 'filters'],
+    'block_specs': [
+        ('convbn', 3, 2, 32),
+        ('depsepconv', 3, 1, 64),
+        ('depsepconv', 3, 2, 128),
+        ('depsepconv', 3, 1, 128),
+        ('depsepconv', 3, 2, 256),
+        ('depsepconv', 3, 1, 256),
+        ('depsepconv', 3, 2, 512),
+        ('depsepconv', 3, 1, 512),
+        ('depsepconv', 3, 1, 512),
+        ('depsepconv', 3, 1, 512),
+        ('depsepconv', 3, 1, 512),
+        ('depsepconv', 3, 1, 512),
+        ('depsepconv', 3, 2, 1024),
+        ('depsepconv', 3, 1, 1024),
+    ]
+}
+
+"""
+Architecture: https://arxiv.org/abs/1801.04381
+
+"MobileNetV2: Inverted Residuals and Linear Bottlenecks"
+Mark Sandler, Andrew Howard, Menglong Zhu, Andrey Zhmoginov, Liang-Chieh Chen
+"""
+MNV2_BLOCK_SPECS = {
+    'spec_name': 'MobileNetV2',
+    'block_spec_schema': ['block_fn', 'kernel_size', 'strides', 'filters',
+                          'expand_ratio'],
+    'block_specs': [
+        ('convbn', 3, 2, 32, None),
+
+        ('mbconv', 3, 1, 16, 1.),
+
+        ('mbconv', 3, 2, 24, 6.),
```

Comment:
Can you rename all `mbconv` to `invertedbottleneck`? The name `mbconv` was introduced in EfficientNet but never mentioned in MobileNet papers.

Reply:
sure, I did it to be consistent with efficientnet.

Reply:
To align with paper, maybe use `invertedres` for inverted residual block, which is mentioned in the paper.